ui_setup.py

```diff
@@ -44,51 +51,163 @@ def __init__(self,master=None,logger=None):
+            self.lbTeams.insert(tk.END,name)
+        for k,v in self.sim.getSides().items():
+            self.lbSideNames.insert(tk.END,k)
+            self.lbTeamAssignments.insert(tk.END,str(self.sim.getTeamName(k)))
+
+    def updateMapNames(self):
+        self.lbMaps.delete(0,tk.END)
+        for name in self.ldr.getMapIDs():
+            self.lbMaps.insert(tk.END,name)
+    # this function is what happens when you hit button "select map" #moved it up here so that it could be bound to selecting map in BuildUI
+    def selectMap(self, event): 
+        curselection = self.lbMaps.curselection()
+        if len(curselection) > 0:
+            # Reset the sim
+            self.sim.reset()
+
+            # Locate the map, copy and give to sim
+            map_ids = self.ldr.getMapIDs()
+            selected_map = self.ldr.copyMapTemplate(map_ids[curselection[0]])
+            self.sim.setMap(selected_map)
+
+            # Construct the map info string
+            mapInfoString =  "NAME:   "+selected_map.getData('name')+'\n'
+            mapInfoString += "DESC:   "+selected_map.getData('desc')+'\n'
+            mapInfoString += "WIDTH:  "+str(selected_map.getData('width'))+'\n'
+            mapInfoString += "HEIGHT: "+str(selected_map.getData('height'))+'\n'
+            mapInfoString += "TEAMS:  "+str(selected_map.getData('teams'))+'\n'
+            mapInfoString += "AGENTS: "+str(selected_map.getData('agents'))+'\n'
+            mapInfoString += "RANDOMLY PLACED OBJECTS:\n"
+            if 'rand_objects' in selected_map.data:
+                for k,v in selected_map.data['rand_objects'].items():
+                    mapInfoString +='   ID:' + str(k) + ' AMT:'+str(v) + '\n'
+            else:
+                mapInfoString += '   NONE\n'
+            mapInfoString += "HAND-PLACED OBJECTS\n"
+            if 'placed_objects' in selected_map.data:
+                for k,v in selected_map.data['placed_objects'].items():
+                    mapInfoString += '   ID:' + str(k) + ' AMT:' + str(len(v)) + '\n'
+            else:
+                mapInfoString += '   NONE\n'
+
+            # Add string to the info text box
+            self.txtMapInfo.delete('1.0',tk.END)
+            self.txtMapInfo.insert(tk.END,mapInfoString)
+            
+            self.updateTeamNames()
+
+    
+
     def BuildUI(self):
+
+        ## PAGE TITLE
+
+        self.pageTitle = uiLabel(master=self, text = "CONFIGURATION")
+        self.pageTitle.pack(side=tk.TOP)
+
+
+        #######################################################################
+        ## SIM UI
+        #######################################################################
+        self.fmSimUI = uiQuietFrame(master=self)
+        self.fmSimUI.pack(side=tk.BOTTOM,fill=tk.BOTH,expand=True,padx=10,pady=10) #maybe remove padding? maybe remove whole frame, and make button part of master frame?
+        
+
+        #below four lines were replaced by below fifth and sixth line
```

Comment:
The lines that this is referencing could be deleted. If your concerned that there won't be a record of them if they need to be changed back for some reason, GitHub keeps track of previous versions, so they wont be lost.